src/backend/brave_search/brave_client.py

```diff
@@ -0,0 +1,302 @@
+import os
+from enum import StrEnum
+from typing import Dict, Optional
+
+import httpx
+import requests
+from pydantic import BaseModel
+from tenacity import AsyncRetrying, stop_after_attempt, wait_fixed
+
+
+class SafeSearchTypes(StrEnum):
+    OFF = "off"
+    MODERATE = "moderate"
+    STRICT = "strict"
+
+
+class WebSearchResponse(BaseModel):
+    """
+    Web search response model.
+    """
+
+    type: str = "search"
+    discussions: Dict = {}
+    faq: Dict = {}
+    infobox: Dict = {}
+    locations: Dict = {}
+    mixed: Dict = {}
+    news: Dict = {}
+    query: Dict = {}
+    videos: Dict = {}
+    web: Dict = {}
+    summarizer: Dict = {}
+
+
+class BraveApiException(Exception):
+    """
+    Custom exception for Brave API errors.
+    """
+    pass
+
+
+class BraveClient:
+    BASE_URL = "https://api.search.brave.com/res/v1/"
+    WEB_SEARCH_API_ENDPOINT = "web/search"
+    MAX_RESULTS = 20
+    MAX_OFFSET = 9
+    WAIT_RETRY_SECONDS = 2
+    RETRY_ATTEMPTS = 2
+
+    def __init__(self, api_key: Optional[str] = None) -> None:
+        self.api_key = api_key if api_key else os.environ.get("BRAVE_API_KEY", None)
```

Comment:
For environment variables, we need to use the config `Settings()` class now, you can refer to the tavily client for an example.

The settings are class attributes instead of fetched during init, like so:

```
class TavilyInternetSearch(BaseTool):
    NAME = "web_search"
    TAVILY_API_KEY = Settings().tools.tavily.api_key
    POST_RERANK_MAX_RESULTS = 6

    def __init__(self):
        self.client = TavilyClient(api_key=self.TAVILY_API_KEY)
```